pxr/usdValidation/usdUtilsValidators/validators.cpp

```diff
@@ -143,6 +144,101 @@ _FileExtensionValidator(const UsdStagePtr& usdStage) {
     return errors;
 }
 
+static UsdValidationErrorVector
+_GetUsdzPackageErrors(const SdfLayerHandle &rootLayer)
+{
+    const UsdZipFile zipFile = UsdZipFile::Open(rootLayer->GetRealPath().c_str());
+    if (!zipFile)
+    {
+        return {};
+    }
+
+    std::string packagePath = ArSplitPackageRelativePathOuter(rootLayer->GetIdentifier()).first;
+
+    UsdValidationErrorVector errors;
+    for(auto it = zipFile.begin(); it != zipFile.end(); ++it)
+    {
+        const UsdZipFile::FileInfo &fileInfo = it.GetFileInfo();
+        if (fileInfo.compressionMethod != 0)
+        {
+            const std::string &fileName = *it;
+            errors.emplace_back(
+                UsdUtilsValidationErrorNameTokens->compressionDetected,
+                UsdValidationErrorType::Error,
+                UsdValidationErrorSites {
+                    UsdValidationErrorSite(
+                            rootLayer, SdfPath(fileName))
+                },
+                TfStringPrintf(
+                "File '%s' in package '%s' has "
+                "compression. Compression method is '%u', actual size "
+                "is %lu. Uncompressed size is %lu.",
+                    fileName.c_str(), packagePath.c_str(),
+                    fileInfo.compressionMethod,
+                    fileInfo.size, fileInfo.uncompressedSize)
+            );
+        }
+
+        if (fileInfo.dataOffset % 64 != 0)
+        {
+            const std::string &fileName = *it;
+            errors.emplace_back(
+                UsdUtilsValidationErrorNameTokens->byteMisalignment,
+                UsdValidationErrorType::Error,
+                UsdValidationErrorSites {
+                    UsdValidationErrorSite(
+                            rootLayer, SdfPath(fileName))
+                },
+                TfStringPrintf(
+                "File '%s' in package '%s' has an "
+                        "invalid offset %zu.",
+                    fileName.c_str(), packagePath.c_str(),
+                    fileInfo.dataOffset)
+            );
+        }
+    }
+
+    return errors;
+}
+
+static UsdValidationErrorVector
+_RootPackageValidator(const UsdStagePtr& usdStage) {
+    const SdfLayerHandle &rootLayer = usdStage->GetRootLayer();
+    return _GetUsdzPackageErrors(rootLayer);
+}
+
+static UsdValidationErrorVector
+_UsdzPackageValidator(const UsdStagePtr& usdStage) {
+
+    SdfLayerRefPtrVector layers;
+    std::vector<std::basic_string<char>> assets, unresolvedPaths;
+    const SdfLayerHandle &rootLayer = usdStage->GetRootLayer();
+    const SdfAssetPath &path = SdfAssetPath(rootLayer->GetIdentifier());
+
+    UsdUtilsComputeAllDependencies(path, &layers, &assets, &unresolvedPaths,
+                                   nullptr);
```

Comment:
For the purposes of this validation test, you don't need assets and unresolvedPaths, so you can simply pass these as nullptr. Implementation for this handles this appropriately: https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdUtils/dependencies.cpp#L174-L182